### PR TITLE
Vb 449 endpoint

### PR DESCRIPTION
**Pull request by philip-milne**
VB-449 Get available slots

Optional min & max days added to get available sessions endpoint
Endpoints, validation and swagger updated to match impl and spec
Endpoint /session-templates renamed visit-session-templates for clarity
Additional exception handler for TemplateNotFoundException to prevent 500 response
Package names & Swagger doc corrected, Static analysis applied
Dev & Test Config updated for new min & max notice defaults
Draft api spec removed & readme updated - swagger available

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerService.kt [hunk 1]
```diff
@@ -119,11 +121,16 @@ class VisitSchedulerService(
       }
   }
 
-  // convenience method to retrieve all session templates to support development
   fun getSessionTemplates(): List<SessionTemplateDto> {
     return sessionTemplateRepository.findAll().sortedBy { it.startDate }.map { SessionTemplateDto(it) }
   }
 
+  @Throws(TemplateNotFoundException::class)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerService.kt:128]: Out of interest, are the throws annotations just mainly for documentation? Or does intelliJ use them in someway?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionResource.kt [hunk 2]
```diff
@@ -45,8 +46,21 @@ class VisitSessionResource(
...
+      example = "MDI"
+    ) prisonId: String,
+    @RequestParam(value = "min", required = false)
+    @Parameter(
+      description = "Minimum number of days notice from the current date",
+      example = "2"
+    ) min: Long?,
+    @RequestParam(value = "max", required = false)
+    @Parameter(
+      description = "Maximum number of days to book-ahead from the current date",
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionResource.kt:61]: meh..worth mentioning that this defaults if not provided?

Comment (philip-milne) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitSessionResource.kt:61]: Good point, ill update.

**Review by rj-adams — APPROVED** [review]
LGTM

**Pull request merged**